### ISSUE TITLE
[Chatllama] Actor for llama

### ISSUE THREAD

**Issue by Vincent131499** [nebuly-ai/nebullvm#236]
When I was using llama-7B to train actor, the following bug appeared:

Current device used :cuda
Loading
Start Actor Model Pretraining
Traceback (most recent call last):
  File "artifacts/main.py", line 51, in <module>
    actor_trainer.train()
  File "/mnt/lustre02/jiangsu/aispeech/home/gfl18/.conda/envs/py38-llm/lib/python3.8/site-packages/chatllama/rlhf/actor.py", line 373, in train
    est_output = self.model(training_input, attention_mask)
  File "/mnt/lustre02/jiangsu/aispeech/home/gfl18/.conda/envs/py38-llm/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "<@beartype(chatllama.rlhf.actor.ActorModel.forward) at 0x2aff142815e0>", line 51, in forward
  File "/mnt/lustre02/jiangsu/aispeech/home/gfl18/.conda/envs/py38-llm/lib/python3.8/site-packages/chatllama/rlhf/actor.py", line 114, in forward
    model_output = self.model.forward(
  File "/mnt/lustre02/jiangsu/aispeech/home/gfl18/.conda/envs/py38-llm/lib/python3.8/site-packages/chatllama/llama_model.py", line 475, in forward
    logits = self._forward(tokens, attention_mask)
  File "/mnt/lustre02/jiangsu/aispeech/home/gfl18/.conda/envs/py38-llm/lib/python3.8/site-packages/chatllama/llama_model.py", line 497, in _forward
    torch.where(kv_mask == 1, float("-inf"), kv_mask).detach().long()
RuntimeError: expected scalar type double but found long int

My environment is：
python3.8
torch1.11.0+cuda113

Excuse me, is it the torch version?

**Comment by XuyaoWang** [comment]
I meet the same bug.

```
Traceback (most recent call last):
  File "artifacts/main.py", line 51, in <module>
    actor_trainer.train()
  File "/work/home/acm3ojrxpa/miniconda3/envs/pytorch110_dtk2210_py38/lib/python3.8/site-packages/chatllama/rlhf/actor.py", line 373, in train
    est_output = self.model(training_input, attention_mask)
  File "/work/home/acm3ojrxpa/miniconda3/envs/pytorch110_dtk2210_py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "<@beartype(chatllama.rlhf.actor.ActorModel.forward) at 0x2b90e7d491f0>", line 51, in forward
  File "/work/home/acm3ojrxpa/miniconda3/envs/pytorch110_dtk2210_py38/lib/python3.8/site-packages/chatllama/rlhf/actor.py", line 114, in forward
    model_output = self.model.forward(
  File "/work/home/acm3ojrxpa/miniconda3/envs/pytorch110_dtk2210_py38/lib/python3.8/site-packages/chatllama/llama_model.py", line 475, in forward
    logits = self._forward(tokens, attention_mask)
  File "/work/home/acm3ojrxpa/miniconda3/envs/pytorch110_dtk2210_py38/lib/python3.8/site-packages/chatllama/llama_model.py", line 497, in _forward
    torch.where(kv_mask == 1, float("-inf"), kv_mask).detach().long()
RuntimeError: expected scalar type double but found long int
```

My environment is:
python    3.8.16
torch       1.10.0a0+git2040069.dtk2210

**Comment by XuyaoWang** [comment]
change code 
`torch.where(kv_mask == 1, float("-inf"), kv_mask).detach().long()`
to
`torch.where(kv_mask == 1, kv_mask.new_tensor(-9223372036854775808), kv_mask).detach().long()`
could sovle this bug

**Comment by PierpaoloSorbellini** [comment]
Hi @XuyaoWang, thanks for the quick fix! Would you mind opening a PR and contributing to the repo by fixing this issue? It would be much appreciated.  :)

**Comment by diegofiori** [comment]
Fixed in #233. Feel free to open another issue if you face any problem!
